### DISCUSSION TITLE
FNIRSI GC-01: USB drive not detected

### DISCUSSION THREAD

**Discussion by offroadbaddude** [Gissio/radpro#38] [Q&A] [answered]
hi ! i've tried to install Gissio's radpro in 2 different fnirsi gc-01 equiped with the j321 gm tube and th ch processor on 3 different pc and 7 different cables but i wasn't able the make the usb drive pop , do i need some kind of driver (windows 10) or is there a feature to enable in the device itself i also tried connecting it with the setting button press down

**Comment by Gissio** [comment, marked as answer]
Some people reported that it didn't work on USB 3.0 ports and that they needed a USB 2.0 port.

Some other people also mentioned that they needed to press the settings key while pressing the power key to enable the USB drive.

Let us know if/how you fixed it, so other users can learn from your experience.

**Reply by offroadbaddude** [reply]
still no luck for me

**Comment by StellaBeatrice** [comment]
![obraz_2024-03-27_212224932](https://github.com/Gissio/radpro/assets/165211472/73b64db7-d269-4435-876f-bbe042e4623c)

https://github.com/Gissio/radpro/assets/165211472/a003955d-e4ea-408c-aee3-3ab5b1f72b9f

> hi ! i've tried to install Gissio's radpro in 2 different fnirsi gc-01 equiped with the j321 gm tube and th ch processor on 3 different pc and 7 different cables but i wasn't able the make the usb drive pop , do i need some kind of driver (windows 10) or is there a feature to enable in the device itself i also tried connecting it with the setting button press down

i have same issue here. Changing ports and pressing both keys didnt work for me

**Reply by dc1rdb** [reply]
Similar issue here. GC-01 manufactured Nov 2023 with Geehy APM32F103RB.
With the factory firmware, my GC-01 went into USB drive mode by pressing and holding the power button and I was able to install RadPro 2.0rc2. 
RadPro 2.0rc2 works well on my GC-01, but I can't get into USB drive mode anymore, no matter what I try (including both batteries removed and reset press). 
Is there any way to reprogram it, maybe via the pin header above the CPU?

![IMG_20240328_065033](https://github.com/Gissio/radpro/assets/9314230/9970a797-1298-4934-8b65-1cc6207d88b5)

**Reply by StellaBeatrice** [reply]
I think that RadPro have special option in setting to enable usb drive mode

**Reply by alfmck** [reply]
I noticed that in order to have connection to USB you have to "kill" previous version installed. To do this I connect GC-01 to PC with win7. It recognizes USB. Then I load two different software's one by one. After that I receive an error. Then I restart device with reset button and connect device to PC with win10. Then USB is recognized, and you can flash the soft you wanted.

**Reply by Gissio** [reply]
dc1rdb: did you have any success? If you did not, please draft a [security advisory](https://github.com/Gissio/radpro/security/advisories) with your email address so I can contact you. Your personal data won't be published.

StellaBeatrice: USB mode if for datalogging, not for flashing the firmware.

**Reply by Gissio** [reply]
Did you try this?

https://github.com/Gissio/radpro/discussions/31#discussioncomment-8845664

**Comment by Lion22332** [comment]
> I noticed that in order to have connection to USB you have to "kill" previous version installed. To do this I connect GC-01 to PC with win7. It recognizes USB. Then I load two different software's one by one. After that I receive an error. Then I restart device with reset button and connect device to PC with win10. Then USB is recognized, and you can flash the soft you wanted. 

Great find! What are the two different software's that you load? Do you drag or copy them to a popup folder? After that you get an error: what doe it say?

**Reply by alfmck** [reply]
I loaded software's suited for my type of microcontroller (ch32f103r8t6).It was 2.0rc2, also original  fnirsi versions.  Not loaded bootloader . No matter you drag or copy. My PC with win7 pc has USB2 ports only but it recognizes GC-01 after switching on, (if device is in hung state you should press reset button). For flashing software, I use PC with Win11 and no matter is GC-01 is connected to USB2 0r USB3. Adding illustrating files.
[GC-01 connected to PC with win11.docx](https://github.com/Gissio/radpro/files/14812904/GC-01.connected.to.PC.with.win11.docx)
[gc-01 loadings with win7.docx](https://github.com/Gissio/radpro/files/14812913/gc-01.loadings.with.win7.docx)

**Reply by Gissio** [reply]
alfmck: how did you flash Rad Pro, if you weren't able to access the bootloader's USB drive?

**Comment by alfmck** [comment]
Jes I tried on ch32f103r8t6 v1.5 loaded from our firmware site. I was unable to connect to usb drive by pressing switch on button with win11. If pressing is wery short - nothing happens; If pressing is a bit longer - gc-01 simply starts working.Sent from my iPhoneOn 30 Mar 2024, at 22:47, Gissio ***@***.***> wrote:﻿
Did you try this?
#31 (comment)

**Comment by alfmck** [comment]
I’m getting ability to access bootloader by connecting GC-01 to win7 and “killing” software in it. After that I reset GC-01, connect it to win11 and switch onSent from my iPhoneOn 2 Apr 2024, at 10:30, Gissio ***@***.***> wrote:﻿
alfmck: how did you flash Rad Pro, if you weren't able to access the bootloader's USB drive?

**Reply by alfmck** [reply]
I tested one more GC-01 bought a year ago in USA. The soft is original, written by supplier. ch32f103r8t6 and J321 used.
When connecting to PC win11 usb3, after short or long press of POWER ON button, the connecting window pops up.
After disconnecting from PC and pressing POWER ON button, the GC-01 does not start. In order to start GC-01 you should press and hold LEFT button and then press POWER ON button.

**Comment by Gissio** [comment]
I've added instructions for flashing Rad Pro to the FNIRSI GC-01 with an STLINK USB dongle: https://github.com/Gissio/radpro/blob/main/docs/devices/FNIRSI%20GC-01/install-stlink.md

**Comment by andrew030977** [comment]
I saw someone connect the gc01 to an android device with a type c to type c cable. The android wont be able to read the gc01 but it sort of active the usb function of it. once the red led is on and the scene stays black, congrats, unplug the android device, and now you can plug it back to your pc usb 2.0 or 3.0 and install the firmware

**Comment by Vteriaza** [comment]
Hola, yo he tenido un problema parecido.
intente instalar el nuevo firmware y me equivoque al  hacerlo, y dejo de verse la unidad disco usb. no reconocia usb. 
tenia un problerma grande ya que ni reconocia usb ni arrancaba el dispositivo.
al dia siguiente , se me ocurrio abrirlo y quitarle la bateria. sin bateria conecte el dispositivo al pc con win10,y aparecio la unidad de disco,a la cual le puse el archivo de el nuevo firmware y milagrosamente arranco de nuevo con el nuevo firmware instalado, no se si te podria valer de algo. saludos.
